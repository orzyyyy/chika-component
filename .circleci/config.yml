version: 2

references:
  container_config: &container_config
    docker:
      - image: circleci/node:11.0.0
    working_directory: ~/chika-component

  attach_workspace: &attach_workspace
    attach_workspace:
      at: ~/chika-component

jobs:
  setup:
    <<: *container_config
    steps:
      - checkout
      - run: node -v
      - run: npm -v
      - run: npm install
      - run:
          command: |
            set +eo
            npm ls
            true
      - persist_to_workspace:
          root: ~/chika-component
          paths:
            - node_modules
  lint:
    <<: *container_config
    steps:
      - checkout
      - *attach_workspace
      - run: npm run lint
  test:
    <<: *container_config
    steps:
      - checkout
      - *attach_workspace
      - run: npm run codecov
      - run: bash <(curl -s https://codecov.io/bash) -t 23446e71-0065-4582-b354-266b384f3625
  compile:
    <<: *container_config
    steps:
      - checkout
      - *attach_workspace
      - run: npm run compile
      - run: ls lib
      - run: ls lib/component
      - run: node ./scripts/dekko-lib.js
      - persist_to_workspace:
          root: ~/chika-component
          paths:
            - lib
  test_lib:
    <<: *container_config
    steps:
      - checkout
      - *attach_workspace
      - run:
          command: npm test -- --clearCache -u
          environment:
            LIB_DIR: lib

workflows:
  version: 2
  build:
    jobs:
      - setup
      - test:
          requires:
            - setup
      - lint:
          requires:
            - setup
      - compile:
          requires:
            - setup
      - test_lib:
          requires:
            - compile
