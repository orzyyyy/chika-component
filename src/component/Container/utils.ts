export const formatConfig = (config: Array<any>) => {
  let result: any = [];
  for (let item of config) {
    const {
      fname,
      controltype,
      fvalue,
      isvisiable,
      isadd,
      defaultvalue,
      editpageorderid,
      isnull,
      issearchfield,
      maxlen,
      minlen,
      foreigndata,
    } = item;
    result.push({
      type: (controlTypeEnums as any)[controltype],
      key: fname,
      name: fvalue,
      showInEdit: isadd,
      showInDetail: isvisiable,
      defaultValue: defaultvalue,
      orderInEdit: editpageorderid,
      orderInDetail: editpageorderid,
      isNull: isnull,
      isSearchItem: issearchfield,
      maxLength: maxlen,
      minLength: minlen,
      foreignData: foreigndata,
    });
  }
  return result;
};

const controlTypeEnums = {
  1: 'input',
  2: 'datePicker',
  3: 'select',
  5: 'checkbox',
  9: 'calendar',
  12: 'upload',
  14: 'mapPicker',
  99: 'label',
};

export const formatControls = (
  dataItem: any,
  config: Array<any>,
  primaryKey: string,
) => {
  let result = [];
  const keys = Object.keys(dataItem);
  for (let item of keys) {
    const targetItem = config.filter(target => target.key === item);
    if (targetItem.length) {
      const { key, showInDetail, type } = targetItem[0];
      if (!showInDetail) continue;
      const value = dataItem[key];
      const item = {
        ...targetItem[0],
        value: dataItem[key],
        templateOrder: dataItem.templateOrder,
        // when mapPicker change, dataSource will change target item by this.
        primaryValue: dataItem[primaryKey],
      };

      // handle with mapPicker
      if (type === 'mapPicker') {
        const latng = value.split('|');
        result.push({
          ...item,
          lng: latng[0],
          lat: latng[1],
          address: latng[2],
        });
      } else {
        result.push(item);
      }
    }
  }
  return result;
};
