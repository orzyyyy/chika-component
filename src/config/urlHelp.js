define([], function () {
  var urlHelp = {
    isReal: false,
    real: {
      search: 'http://222.185.24.19:9002/webapi/api/v2/generalbackstage/getdata',
      getConfig: 'http://222.185.24.19:9002/webapi/api/v2/generalbackstage/getconfig',
      getUrlData: 'http://222.185.24.19:9002/webapi/api/v2/generalbackstage/getinterfacedata', // 第三方接口地址，为了放置跨域问题，后端先处理下
      generalbackstage: 'http://222.185.24.19:9002/webapi/api/v2/generalbackstage/operatedata', // 增删改
    },
    simulation: {
      tableconfig: './data/tableconfig.json',
      tableDatas: './data/tableDatas.json',
      newTabDatas: './data/newTabDatas.json',
      builder_search: '../../CommonReport/AjaxHandler/BuilderHandler.ashx?OPType=Search',
      getConfig: './data/getConfig.json',
      search: './data/search.json',
      calendar_demo: './data/calendar_demo.json',
      secRecordPath: './data/s_problem_record_hy.json',
      getUrlData: './data/tableconfig.json', // 第三方接口地址，为了防止跨域问题，后端先处理下
      generalbackstage: './data/tableconfig.json', // 增删改
    }
  }
  return urlHelp
});