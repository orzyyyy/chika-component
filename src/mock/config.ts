export const originConfig = [
  {
    id: 113442,
    fname: 'pjnm',
    fvalue: 'pjnm',
    physicalcolname: '',
    iskey: false,
    isnull: false,
    issearchfield: true,
    searchtype: -1,
    searchorder: null,
    minlen: 0,
    maxlen: 5,
    decimalcount: null,
    regular: '',
    orderid: 26,
    isvisiable: true,
    defaultvalue: '',
    unit: '',
    controltype: 99,
    editpageorderid: 21,
    controlheight: '',
    controlwidth: '',
    isadd: true,
    fieldtag: null,
    foreigndata: [],
    dateformat: 'yyyy-MM-dd HH:mm:ss',
    fieldpar: '',
  },
  {
    id: 113438,
    fname: 'dam_cd',
    fvalue: 'dam_cd',
    physicalcolname: 'dam_cd',
    iskey: true,
    isnull: false,
    issearchfield: true,
    searchtype: 8,
    searchorder: null,
    minlen: null,
    maxlen: 8,
    decimalcount: null,
    regular: '',
    orderid: 1,
    isvisiable: true,
    defaultvalue: '',
    unit: '',
    controltype: 1,
    editpageorderid: 1,
    controlheight: '',
    controlwidth: '',
    isadd: true,
    fieldtag: null,
    foreigndata: [],
    dateformat: 'yyyy-MM-dd HH:mm:ss',
    fieldpar: '',
    isSort: true,
  },
  {
    id: 113442,
    fname: 'datePicker',
    fvalue: 'datePicker',
    physicalcolname: 'attachment',
    iskey: false,
    isnull: false,
    issearchfield: false,
    searchtype: 5,
    searchorder: null,
    minlen: null,
    maxlen: 50,
    decimalcount: null,
    regular: '',
    orderid: 21,
    isvisiable: true,
    defaultvalue: '',
    unit: '',
    controltype: 2,
    editpageorderid: 16,
    controlheight: '',
    controlwidth: '',
    isadd: true,
    fieldtag: null,
    foreigndata: [],
    dateformat: 'yyyy-MM-dd HH:mm:ss',
  },
  {
    id: 113442,
    fname: 'select',
    fvalue: 'select',
    physicalcolname: 'attachment',
    iskey: false,
    isnull: false,
    issearchfield: false,
    searchtype: 5,
    searchorder: null,
    minlen: null,
    maxlen: 8,
    decimalcount: null,
    regular: '',
    orderid: 21,
    isvisiable: true,
    defaultvalue: '',
    unit: '',
    controltype: 3,
    editpageorderid: 16,
    controlheight: '',
    controlwidth: '',
    isadd: true,
    fieldtag: null,
    foreigndata: [
      { label: 'label1', value: 'value1' },
      { label: 'label2', value: 'value2' },
    ],
    dateformat: 'yyyy-MM-dd HH:mm:ss',
  },
  {
    id: 113442,
    fname: 'checkbox',
    fvalue: 'checkbox',
    physicalcolname: 'attachment',
    iskey: false,
    isnull: false,
    issearchfield: false,
    searchtype: 5,
    searchorder: null,
    minlen: null,
    maxlen: 8,
    decimalcount: null,
    regular: '',
    orderid: 21,
    isvisiable: true,
    defaultvalue: '',
    unit: '',
    controltype: 5,
    editpageorderid: 16,
    controlheight: '',
    controlwidth: '',
    isadd: true,
    fieldtag: null,
    foreigndata: [
      { label: 'checkboxlable1', value: 'checkbox1' },
      { label: 'label2', value: 'checkbox2' },
      { label: 'label3', value: 'checkbox3' },
    ],
    dateformat: 'yyyy-MM-dd HH:mm:ss',
  },
  {
    id: 113442,
    fname: 'calendar',
    fvalue: 'calendar',
    physicalcolname: 'attachment',
    iskey: false,
    isnull: false,
    issearchfield: false,
    searchtype: 5,
    searchorder: null,
    minlen: null,
    maxlen: 8,
    decimalcount: null,
    regular: '',
    orderid: 21,
    isvisiable: true,
    defaultvalue: '',
    unit: '',
    controltype: 9,
    editpageorderid: 16,
    controlheight: '',
    controlwidth: '',
    isadd: true,
    fieldtag: null,
    foreigndata: [],
    dateformat: 'yyyy-MM-dd HH:mm:ss',
  },
  {
    id: 113442,
    fname: 'upload',
    fvalue: 'upload',
    physicalcolname: 'attachment',
    iskey: false,
    isnull: false,
    issearchfield: false,
    searchtype: 5,
    searchorder: null,
    minlen: null,
    maxlen: 8,
    decimalcount: 2,
    regular: '',
    orderid: 21,
    isvisiable: true,
    defaultvalue: '',
    unit: '',
    controltype: 12,
    editpageorderid: 16,
    controlheight: '',
    controlwidth: '',
    isadd: true,
    fieldtag: null,
    foreigndata: [],
    dateformat: 'yyyy-MM-dd HH:mm:ss',
  },
  {
    id: 113442,
    fname: 'mapPicker',
    fvalue: 'mapPicker',
    physicalcolname: 'attachment',
    iskey: false,
    isnull: false,
    issearchfield: false,
    searchtype: 5,
    searchorder: null,
    minlen: null,
    maxlen: 8,
    decimalcount: 2,
    regular: '',
    orderid: 21,
    isvisiable: true,
    defaultvalue: '',
    unit: '',
    controltype: 14,
    editpageorderid: 16,
    controlheight: '',
    controlwidth: '',
    isadd: true,
    fieldtag: null,
    foreigndata: [],
    dateformat: 'yyyy-MM-dd HH:mm:ss',
  },
];
