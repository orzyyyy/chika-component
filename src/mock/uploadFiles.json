{
  "data": {
    "list": [
      {
        "id": 1,
        "annex": "[{\"filepath\":\"../../assets/d19f18a8ly1g0akzu7cq5j20u016gnpe.jpg\",\"filetile\":\"name0\",\"filename\":\"xls\"},{\"filepath\":\"../../assets/d19f18a8ly1g0akzu7cq5j20u016gnpe.jpg\",\"filetile\":\"name1\",\"filename\":\"doc\"}]"
      },
      {
        "id": 2,
        "annex": "[{\"filepath\":\"../../assets/d19f18a8ly1g0akzu7cq5j20u016gnpe.jpg\",\"filetile\":\"name2\",\"filename\":\"jpg\"}]"
      }
    ],
    "recordcount": 3
  }
}
