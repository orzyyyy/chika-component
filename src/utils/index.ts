import { format as fnsFormat } from 'date-fns/esm';
import { zhCN } from 'date-fns/locale';

export const formatDate = (
  date: Date | string = new Date(),
  format: string = 'yyyy-MM-dd HH:mm:ss',
) => {
  if (typeof date === 'string') {
    date = new Date(date);
  }
  return fnsFormat(date, format, {
    locale: zhCN,
  });
};

export const getParamsFromUrl = (name: string, url?: string) => {
  const targetUrl = url || window.location.hash;
  const reg = new RegExp('(^|&)' + name + '=([^&]*)(&|$)');
  let result: any =
    targetUrl.split('?').length > 1 ? targetUrl.split('?')[1].match(reg) : 0;
  if (result) {
    return result[2];
  }
  return null;
};

export const compare = (property: any, isDesc?: boolean) =>
  isDesc
    ? (b: any, a: any) => a[property] - b[property]
    : (a: any, b: any) => a[property] - b[property];

export const getDirection = (
  startX: number,
  startY: number,
  endX: number,
  endY: number,
) => {
  let angx = endX - startX;
  let angy = endY - startY;

  if (Math.abs(angx) < 25 && Math.abs(angy) < 25) {
    return '';
  }

  let angle = (Math.atan2(angy, angx) * 180) / Math.PI;

  if (angle >= -135 && angle <= -45) {
    return 'toTop';
  } else if (angle > 45 && angle < 135) {
    return 'toDown';
  } else if (
    (angle >= 135 && angle <= 180) ||
    (angle >= -180 && angle < -135)
  ) {
    return 'toLeft';
  } else if (angle >= -45 && angle <= 45) {
    return 'toRight';
  }
  return '';
};

export const bindTouchDirection = (
  ref: any,
  callback: (direction: string) => void,
) => {
  if (!ref) {
    return callback('error');
  }
  let startX: number;
  let startY: number;
  let endX: number;
  let endY: number;

  ref.addEventListener('touchstart', (e: any) => {
    startX = e.touches[0].pageX;
    startY = e.touches[0].pageY;
  });

  ref.addEventListener('touchend', (e: any) => {
    endX = e.changedTouches[0].pageX;
    endY = e.changedTouches[0].pageY;

    callback(getDirection(startX, startY, endX, endY));
  });
};
