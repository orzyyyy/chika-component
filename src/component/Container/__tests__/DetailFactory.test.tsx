import React from 'react';
import { mount, shallow } from 'enzyme';
import 'nino-cli/scripts/setup';
import { originConfig } from '../../../mock/config';
import { originDataSource } from '../../../mock/dataSource';
const DataItem = originDataSource[0];
import { formatControls, formatConfig } from '../utils';
const dataItem = formatControls(DataItem, formatConfig(originConfig), 'dam_cd');
let DetailFactory;
switch (process.env.LIB_DIR) {
  case 'lib':
    DetailFactory = require('../../../../lib/component/Container/DetailFactory')
      .default;
    break;
  default:
    DetailFactory = require('../DetailFactory').default;
    break;
}

const handleChildDataFormat = (
  value: string | number | Date,
  childProps: any,
  bindKey: string,
) => {
  for (let item of formatConfig(originConfig)) {
    const { key, unit, decimalCount } = item;
    if (key === childProps[bindKey]) {
      if (value instanceof Date) {
        // todo: find a way not to transform local to utc
        // for ci problem, local can't get reproduce
        // value = formatDate(value, dateFormat);
        value = '2017-08-09 08:00:00';
      }
      if (decimalCount) {
        value = +parseFloat(
          parseFloat(value.toString()).toFixed(decimalCount),
        ).toPrecision(12);
      }
      if (unit) {
        value = `${value} ${unit}`;
      }
    }
  }
  return value;
};

describe('DetailFactory', () => {
  it('render correctly', () => {
    const wrapper = shallow(
      <DetailFactory
        dataSource={dataItem}
        onDataFormat={handleChildDataFormat}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('onMapBoxChange should work correctly', () => {
    const onMapBoxChange = jest.fn();
    const wrapper = mount(
      <DetailFactory
        dataSource={dataItem}
        onMapBoxChange={onMapBoxChange}
        onDataFormat={handleChildDataFormat}
      />,
    );
    wrapper
      .find('ListItem')
      .at(5)
      .simulate('click');
    expect(onMapBoxChange).toBeCalled();
  });
});
