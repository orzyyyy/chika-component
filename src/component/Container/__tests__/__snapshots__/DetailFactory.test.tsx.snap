// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailFactory render correctly 1`] = `
<DetailFactory
  dataSource={
    Array [
      Object {
        "dateFormat": "yyyy-MM-dd HH:mm:ss",
        "decimalCount": null,
        "defaultValue": "",
        "foreignData": Array [],
        "id": 113438,
        "isNull": false,
        "isSearchItem": true,
        "isSort": true,
        "key": "dam_cd",
        "maxLength": 8,
        "minLength": null,
        "name": "dam_cd",
        "orderInDetail": 1,
        "orderInEdit": 1,
        "primaryValue": "Z000001",
        "showInDetail": true,
        "showInEdit": true,
        "templateOrder": undefined,
        "type": "input",
        "unit": "",
        "value": "Z000001",
      },
      Object {
        "dateFormat": "yyyy-MM-dd HH:mm:ss",
        "decimalCount": null,
        "defaultValue": "",
        "foreignData": Array [],
        "id": 113442,
        "isNull": false,
        "isSearchItem": true,
        "isSort": undefined,
        "key": "pjnm",
        "maxLength": 5,
        "minLength": 0,
        "name": "pjnm",
        "orderInDetail": 21,
        "orderInEdit": 21,
        "primaryValue": "Z000001",
        "showInDetail": true,
        "showInEdit": true,
        "templateOrder": undefined,
        "type": "label",
        "unit": "",
        "value": "德江坞",
      },
      Object {
        "address": "杭州1",
        "dateFormat": "yyyy-MM-dd HH:mm:ss",
        "decimalCount": 2,
        "defaultValue": "",
        "foreignData": Array [],
        "id": 113442,
        "isNull": false,
        "isSearchItem": false,
        "isSort": undefined,
        "key": "mapPicker",
        "lat": "39.89520118467971",
        "lng": "116.30038976669313",
        "maxLength": 8,
        "minLength": null,
        "name": "mapPicker",
        "orderInDetail": 16,
        "orderInEdit": 16,
        "primaryValue": "Z000001",
        "showInDetail": true,
        "showInEdit": true,
        "templateOrder": undefined,
        "type": "mapPicker",
        "unit": "",
        "value": "116.30038976669313|39.89520118467971|杭州1",
      },
      Object {
        "dateFormat": "yyyy-MM-dd HH:mm:ss",
        "decimalCount": null,
        "defaultValue": "",
        "foreignData": Array [],
        "id": 113442,
        "isNull": false,
        "isSearchItem": false,
        "isSort": undefined,
        "key": "datePicker",
        "maxLength": 50,
        "minLength": null,
        "name": "datePicker",
        "orderInDetail": 16,
        "orderInEdit": 16,
        "primaryValue": "Z000001",
        "showInDetail": true,
        "showInEdit": true,
        "templateOrder": undefined,
        "type": "datePicker",
        "unit": "",
        "value": 2017-08-09T08:00:00.000Z,
      },
      Object {
        "dateFormat": "yyyy-MM-dd HH:mm:ss",
        "decimalCount": null,
        "defaultValue": "",
        "foreignData": Array [
          Object {
            "label": "label1",
            "value": "value1",
          },
          Object {
            "label": "label2",
            "value": "value2",
          },
        ],
        "id": 113442,
        "isNull": false,
        "isSearchItem": false,
        "isSort": undefined,
        "key": "select",
        "maxLength": 8,
        "minLength": null,
        "name": "select",
        "orderInDetail": 16,
        "orderInEdit": 16,
        "primaryValue": "Z000001",
        "showInDetail": true,
        "showInEdit": true,
        "templateOrder": undefined,
        "type": "select",
        "unit": "",
        "value": "value1",
      },
      Object {
        "dateFormat": "yyyy-MM-dd HH:mm:ss",
        "decimalCount": null,
        "defaultValue": "",
        "foreignData": Array [
          Object {
            "label": "checkboxlable1",
            "value": "checkbox1",
          },
          Object {
            "label": "label2",
            "value": "checkbox2",
          },
          Object {
            "label": "label3",
            "value": "checkbox3",
          },
        ],
        "id": 113442,
        "isNull": false,
        "isSearchItem": false,
        "isSort": undefined,
        "key": "checkbox",
        "maxLength": 8,
        "minLength": null,
        "name": "checkbox",
        "orderInDetail": 16,
        "orderInEdit": 16,
        "primaryValue": "Z000001",
        "showInDetail": true,
        "showInEdit": true,
        "templateOrder": undefined,
        "type": "checkbox",
        "unit": "",
        "value": "checkbox1",
      },
      Object {
        "dateFormat": "yyyy-MM-dd HH:mm:ss",
        "decimalCount": null,
        "defaultValue": "",
        "foreignData": Array [],
        "id": 113442,
        "isNull": false,
        "isSearchItem": false,
        "isSort": undefined,
        "key": "calendar",
        "maxLength": 8,
        "minLength": null,
        "name": "calendar",
        "orderInDetail": 16,
        "orderInEdit": 16,
        "primaryValue": "Z000001",
        "showInDetail": true,
        "showInEdit": true,
        "templateOrder": undefined,
        "type": "calendar",
        "unit": "",
        "value": Array [
          "2018-04-03 00:00:00",
          "2018-04-04 00:00:00",
        ],
      },
      Object {
        "dateFormat": "yyyy-MM-dd HH:mm:ss",
        "decimalCount": 2,
        "defaultValue": "",
        "foreignData": Array [],
        "id": 113442,
        "isNull": false,
        "isSearchItem": false,
        "isSort": undefined,
        "key": "upload",
        "maxLength": 8,
        "minLength": null,
        "name": "upload",
        "orderInDetail": 16,
        "orderInEdit": 16,
        "primaryValue": "Z000001",
        "showInDetail": true,
        "showInEdit": true,
        "templateOrder": undefined,
        "type": "upload",
        "unit": "",
        "value": Array [
          Object {
            "id": 1,
            "name": "name2",
            "url": "../../assets/d19f18a8ly1g0akzu7cq5j20u016gnpe.jpg",
          },
        ],
      },
    ]
  }
  onDataFormat={[Function]}
>
  <div
    className="DetailFactory"
  >
    <DetailArrow
      showLast={false}
      showNext={false}
    />
    <List
      prefixCls="am-list"
    >
      <div
        className="am-list"
      >
        <div
          className="am-list-body"
        >
          <ListItem
            align="middle"
            error={false}
            multipleLine={false}
            platform="ios"
            prefixCls="am-list"
            wrap={false}
          >
            <TouchFeedback
              activeClassName="am-list-item-active"
              disabled={true}
            >
              <div
                className="am-list-item am-list-item-middle"
                onClick={[Function]}
              >
                <div
                  className="am-list-line"
                >
                  <div
                    className="am-list-content"
                  >
                    <ListItem
                      align="middle"
                      error={false}
                      extra="Z000001"
                      key="detail-page-label-0"
                      multipleLine={false}
                      platform="ios"
                      prefixCls="am-list"
                      wrap={false}
                    >
                      <TouchFeedback
                        activeClassName="am-list-item-active"
                        disabled={true}
                      >
                        <div
                          className="am-list-item am-list-item-middle"
                          onClick={[Function]}
                        >
                          <div
                            className="am-list-line"
                          >
                            <div
                              className="am-list-content"
                            >
                              dam_cd
                            </div>
                            <div
                              className="am-list-extra"
                            >
                              Z000001
                            </div>
                          </div>
                          <div
                            className="am-list-ripple"
                            style={
                              Object {
                                "display": "none",
                              }
                            }
                          />
                        </div>
                      </TouchFeedback>
                    </ListItem>
                    <ListItem
                      align="middle"
                      error={false}
                      extra="德江坞"
                      key="detail-page-label-1"
                      multipleLine={false}
                      platform="ios"
                      prefixCls="am-list"
                      wrap={false}
                    >
                      <TouchFeedback
                        activeClassName="am-list-item-active"
                        disabled={true}
                      >
                        <div
                          className="am-list-item am-list-item-middle"
                          onClick={[Function]}
                        >
                          <div
                            className="am-list-line"
                          >
                            <div
                              className="am-list-content"
                            >
                              pjnm
                            </div>
                            <div
                              className="am-list-extra"
                            >
                              德江坞
                            </div>
                          </div>
                          <div
                            className="am-list-ripple"
                            style={
                              Object {
                                "display": "none",
                              }
                            }
                          />
                        </div>
                      </TouchFeedback>
                    </ListItem>
                    <ListItem
                      align="middle"
                      error={false}
                      extra="116.30038976669313"
                      multipleLine={false}
                      platform="ios"
                      prefixCls="am-list"
                      wrap={false}
                    >
                      <TouchFeedback
                        activeClassName="am-list-item-active"
                        disabled={true}
                      >
                        <div
                          className="am-list-item am-list-item-middle"
                          onClick={[Function]}
                        >
                          <div
                            className="am-list-line"
                          >
                            <div
                              className="am-list-content"
                            >
                              经度
                            </div>
                            <div
                              className="am-list-extra"
                            >
                              116.30038976669313
                            </div>
                          </div>
                          <div
                            className="am-list-ripple"
                            style={
                              Object {
                                "display": "none",
                              }
                            }
                          />
                        </div>
                      </TouchFeedback>
                    </ListItem>
                    <ListItem
                      align="middle"
                      error={false}
                      extra="39.89520118467971"
                      multipleLine={false}
                      platform="ios"
                      prefixCls="am-list"
                      wrap={false}
                    >
                      <TouchFeedback
                        activeClassName="am-list-item-active"
                        disabled={true}
                      >
                        <div
                          className="am-list-item am-list-item-middle"
                          onClick={[Function]}
                        >
                          <div
                            className="am-list-line"
                          >
                            <div
                              className="am-list-content"
                            >
                              纬度
                            </div>
                            <div
                              className="am-list-extra"
                            >
                              39.89520118467971
                            </div>
                          </div>
                          <div
                            className="am-list-ripple"
                            style={
                              Object {
                                "display": "none",
                              }
                            }
                          />
                        </div>
                      </TouchFeedback>
                    </ListItem>
                    <ListItem
                      align="middle"
                      arrow="horizontal"
                      error={false}
                      extra="杭州1"
                      multipleLine={false}
                      onClick={[Function]}
                      platform="ios"
                      prefixCls="am-list"
                      wrap={false}
                    >
                      <TouchFeedback
                        activeClassName="am-list-item-active"
                        disabled={false}
                      >
                        <div
                          className="am-list-item am-list-item-middle"
                          onClick={[Function]}
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchCancel={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                        >
                          <div
                            className="am-list-line"
                          >
                            <div
                              className="am-list-content"
                            >
                              地址
                            </div>
                            <div
                              className="am-list-extra"
                            >
                              杭州1
                            </div>
                            <div
                              aria-hidden="true"
                              className="am-list-arrow am-list-arrow-horizontal"
                            />
                          </div>
                          <div
                            className="am-list-ripple"
                            style={
                              Object {
                                "display": "none",
                              }
                            }
                          />
                        </div>
                      </TouchFeedback>
                    </ListItem>
                    <ListItem
                      align="middle"
                      error={false}
                      extra="2017-08-09 08:00:00"
                      key="detail-page-label-3"
                      multipleLine={false}
                      platform="ios"
                      prefixCls="am-list"
                      wrap={false}
                    >
                      <TouchFeedback
                        activeClassName="am-list-item-active"
                        disabled={true}
                      >
                        <div
                          className="am-list-item am-list-item-middle"
                          onClick={[Function]}
                        >
                          <div
                            className="am-list-line"
                          >
                            <div
                              className="am-list-content"
                            >
                              datePicker
                            </div>
                            <div
                              className="am-list-extra"
                            >
                              2017-08-09 08:00:00
                            </div>
                          </div>
                          <div
                            className="am-list-ripple"
                            style={
                              Object {
                                "display": "none",
                              }
                            }
                          />
                        </div>
                      </TouchFeedback>
                    </ListItem>
                    <ListItem
                      align="middle"
                      error={false}
                      extra="value1"
                      key="detail-page-label-4"
                      multipleLine={false}
                      platform="ios"
                      prefixCls="am-list"
                      wrap={false}
                    >
                      <TouchFeedback
                        activeClassName="am-list-item-active"
                        disabled={true}
                      >
                        <div
                          className="am-list-item am-list-item-middle"
                          onClick={[Function]}
                        >
                          <div
                            className="am-list-line"
                          >
                            <div
                              className="am-list-content"
                            >
                              select
                            </div>
                            <div
                              className="am-list-extra"
                            >
                              value1
                            </div>
                          </div>
                          <div
                            className="am-list-ripple"
                            style={
                              Object {
                                "display": "none",
                              }
                            }
                          />
                        </div>
                      </TouchFeedback>
                    </ListItem>
                    <ListItem
                      align="middle"
                      error={false}
                      extra="checkbox1"
                      key="detail-page-label-5"
                      multipleLine={false}
                      platform="ios"
                      prefixCls="am-list"
                      wrap={false}
                    >
                      <TouchFeedback
                        activeClassName="am-list-item-active"
                        disabled={true}
                      >
                        <div
                          className="am-list-item am-list-item-middle"
                          onClick={[Function]}
                        >
                          <div
                            className="am-list-line"
                          >
                            <div
                              className="am-list-content"
                            >
                              checkbox
                            </div>
                            <div
                              className="am-list-extra"
                            >
                              checkbox1
                            </div>
                          </div>
                          <div
                            className="am-list-ripple"
                            style={
                              Object {
                                "display": "none",
                              }
                            }
                          />
                        </div>
                      </TouchFeedback>
                    </ListItem>
                    <ListItem
                      align="middle"
                      error={false}
                      extra="2018-04-03 00:00:00"
                      multipleLine={false}
                      platform="ios"
                      prefixCls="am-list"
                      wrap={false}
                    >
                      <TouchFeedback
                        activeClassName="am-list-item-active"
                        disabled={true}
                      >
                        <div
                          className="am-list-item am-list-item-middle"
                          onClick={[Function]}
                        >
                          <div
                            className="am-list-line"
                          >
                            <div
                              className="am-list-content"
                            >
                              calendar开始时间
                            </div>
                            <div
                              className="am-list-extra"
                            >
                              2018-04-03 00:00:00
                            </div>
                          </div>
                          <div
                            className="am-list-ripple"
                            style={
                              Object {
                                "display": "none",
                              }
                            }
                          />
                        </div>
                      </TouchFeedback>
                    </ListItem>
                    <ListItem
                      align="middle"
                      error={false}
                      extra="2018-04-04 00:00:00"
                      multipleLine={false}
                      platform="ios"
                      prefixCls="am-list"
                      wrap={false}
                    >
                      <TouchFeedback
                        activeClassName="am-list-item-active"
                        disabled={true}
                      >
                        <div
                          className="am-list-item am-list-item-middle"
                          onClick={[Function]}
                        >
                          <div
                            className="am-list-line"
                          >
                            <div
                              className="am-list-content"
                            >
                              calendar结束时间
                            </div>
                            <div
                              className="am-list-extra"
                            >
                              2018-04-04 00:00:00
                            </div>
                          </div>
                          <div
                            className="am-list-ripple"
                            style={
                              Object {
                                "display": "none",
                              }
                            }
                          />
                        </div>
                      </TouchFeedback>
                    </ListItem>
                  </div>
                </div>
                <div
                  className="am-list-ripple"
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                />
              </div>
            </TouchFeedback>
          </ListItem>
          <ListItem
            align="middle"
            error={false}
            multipleLine={false}
            platform="ios"
            prefixCls="am-list"
            wrap={false}
          >
            <TouchFeedback
              activeClassName="am-list-item-active"
              disabled={true}
            >
              <div
                className="am-list-item am-list-item-middle"
                onClick={[Function]}
              >
                <div
                  className="am-list-line"
                >
                  <div
                    className="am-list-content"
                  >
                    <Button
                      activeStyle={Object {}}
                      disabled={false}
                      inline={false}
                      loading={false}
                      prefixCls="am-button"
                      size="large"
                    >
                      <TouchFeedback
                        activeClassName="am-button-active"
                        activeStyle={Object {}}
                        disabled={false}
                      >
                        <a
                          aria-disabled={false}
                          className="am-button"
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchCancel={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          role="button"
                        >
                          <span
                            key=".0"
                          >
                            返 回
                          </span>
                        </a>
                      </TouchFeedback>
                    </Button>
                  </div>
                </div>
                <div
                  className="am-list-ripple"
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                />
              </div>
            </TouchFeedback>
          </ListItem>
        </div>
      </div>
    </List>
  </div>
</DetailFactory>
`;
